### PR TITLE
Fix #7662: Avatar: Color adjustments for dynamicColor mode

### LINKED ISSUE #7662: Avatar: Color adjustments for dynamicColor mode
**Description**

The current color spectrum of the `dynamicColor` mode might not fit into the theme of a page.

**Describe the solution you would like**

The widget could be used in a more flexible way if it would allow to customize the generated background colors.
How about two new optional attributes for saturation and lightness components?

**Example XHTML**
```xml
<p:avatar dynamicColor="true" saturation="80" lightness="33" label="..." />
```

**AvatarRenderer**
```java
    protected String generateBackgroundColor(Avatar avatar) {
        return "color:#fff;background-color: hsl(" 
                + Math.abs((avatar.getLabel().hashCode() % 40) * 9) + ", " 
                + avatar.getSaturation() + "%, "
                + avatar.getLightness() + "%);";
    }
```

### PR DESCRIPTION
**Pull request by jepsar**
Fix #7662

### CODE REVIEW

File: docs/migrationguide/11_0_0.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ The following components have been removed.
 ## AutoComplete
   * Style class `ui-autocomplete-emptyMessage` renamed to `ui-autocomplete-empty-message`.
 
+## Avatar
+  * Removed dynamic color `mix-blend-mode: difference` style in favour of `ui-avatar-light` and `ui-avatar-light` based on the lightness.
```
Comment (melloware) [inline docs/migrationguide/11_0_0.md]: I think you mean light and "dark" in  the second on here?

File: primefaces/src/main/java/org/primefaces/component/avatar/AvatarRenderer.java [hunk 2]
```diff
@@ -145,11 +145,16 @@ protected String calculateLabel(FacesContext context, Avatar avatar) {
...
      */
-    protected String generateBackgroundColor(String label) {
-        return "color:#fff;background-color: hsl(" + Math.abs((label.hashCode() % 40) * 9) + ", 100%, 50%);";
+    protected String generateBackgroundColor(Avatar avatar) {
+        boolean withAlpha = avatar.getAlpha() != null;
+        return "background-color:hsl" + (withAlpha ? "a" : Constants.EMPTY_STRING) + "(" +
+                Math.abs((avatar.getLabel().hashCode() % 40) * 9) +
+                "," + avatar.getSaturation() + "%" +
+                "," + avatar.getLightness() + "%" +
+                (withAlpha ? "," + avatar.getAlpha() + "%" : Constants.EMPTY_STRING) + ");";
```
Comment (melloware) [inline primefaces/src/main/java/org/primefaces/component/avatar/AvatarRenderer.java]: Just to clean this code up a little maybe you should use `String.format()` to make it easier to read and pass variables to?

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
Nice work!

File: primefaces/src/main/resources/META-INF/resources/primefaces/avatar/avatar.css [hunk 3]
```diff
@@ -24,6 +24,10 @@
     height: 100%;
 }
 
-.ui-avatar .ui-avatar-dynamic {
-    mix-blend-mode: difference;
-}
\ No newline at end of file
+.ui-avatar.ui-avatar-dark {
```
Comment (jepsar) [inline primefaces/src/main/resources/META-INF/resources/primefaces/avatar/avatar.css:27]: @melloware @mertsincan @sqores maybe this styling of dynamic background text colors should go to the themes. WDYT?

Comment (melloware) [reply primefaces/src/main/resources/META-INF/resources/primefaces/avatar/avatar.css:27]: I think for now defaults are OK here and they can be overridden in themes if necessary.

**Pull request merged**